adr/0026-write-unit-tests-with-core-tracker.md

```diff
@@ -0,0 +1,51 @@
+# Event unit tests
+
+ * Status: Accepted
+ * Deciders: iOS Team
+ * Date: 2023-03-08
+
+ ## Context
+
+ With the core tracker structure, we will be able to write key/value event tests correctly.
+
+ ## Decision
+
+ During event testing, it's possible to use the same value across all keys, and the test will still pass as long as the count of key/value matches and the values are included in the sent values. However, this approach may result in incorrect key/value tests passing.
+
+ ✅ test passed
+ ```swift 
+  trackableServiceTest.assertValues(
+     .firebase, ["Homepage",
+                 "Support",
+                 "Account"]
+ )
+ ```
+
+ ✅ test passed
+ ```swift 
+  trackableServiceTest.assertValues(
+     .firebase, ["Homepage",
+                 "Homepage",
+                 "Homepage"]
+ )
+ ```
+
+ After implementing the mockassertable framework, the coretracker began to function correctly, allowing us to conduct more accurate event tests using key-value pairs. Thus, we recommend using coretracker instead of trackableservicetest to obtain precise test results.
```

Comment:
Actually, `MockAssertable` protocol is just a wrapper that uses `CoreTracker` as a backbone. That's why we can correct these statements to make it more precise as following;

After implementing with `MockAssertable` protocol, we started to evaluate event related data more efficiently and precisely from mock services under `CoreTracker`. Therefore, instead of using 'MockCoreTracker' directly, we can get stable test results by injecting 'TrackableServiceTest' into the classes to be tested and testing the values ​​over it.